src/main/java/com/celarli/commons/vfs/provider/google/GCSFileProvider.java

```diff
@@ -43,21 +44,43 @@ protected FileSystem doCreateFileSystem(FileName fileName, FileSystemOptions fil
 
         try {
 
-            InputStream fis = GcsFileSystemConfigBuilder.getInstance().getKeyStream(fileSystemOptions);
+            Storage storage;
 
-            if (fis == null) {
-                throw new FileSystemException("Credential not found");
-            }
+            Integer type = GcsFileSystemConfigBuilder.getInstance().getClientType(fileSystemOptions);
+            ClientType clientType = ClientType.getByType(type);
+            switch (clientType) {
```

Comment:
Rather than using this, I'll introduce a Factory.

Reply:
thanks, I will also push some code with further refactoring